include/aero/array/array.h

```diff
@@ -1,7 +1,58 @@
 #ifndef AERO_ARRAY_H
 #define AERO_ARRAY_H
 
-/// This opaque type represents an array implemented in C.
+#include <aero/aero.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// This opaque type represents an implementation of an Array
+typedef struct aero_array_data_t aero_array_data_t;
+
+/// Behaviors and data associated with an Array. All of these functions
+/// are passed the data pointer created by an Array constructor
 typedef struct aero_array_t aero_array_t;
+typedef struct aero_array_t {
```

Comment:
Did you need to marry the behaviors and array type in order to expose its innards to different contexts? One alternative to this is to stick the definition of the struct itself into a `.h` file within the implementation directory. I don't believe anyone else should need to see this. Maybe we can chat about this.

Reply:
Yeah, maybe we could talk about this. I'm not quite getting what you're proposing to do.